ccl/streamingccl/streamingest: TestTenantStreamingPauseOnPermanentJobError failed

ccl/streamingccl/streamingest.TestTenantStreamingPauseOnPermanentJobError [failed](https://tanzanite.cluster.engflow.com/invocations/default/e28999f8-4d51-4e92-b3d9-a9e402595828?testReportRun=29&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL3N0cmVhbWluZ2NjbC9zdHJlYW1pbmdlc3Q6c3RyZWFtaW5nZXN0X3Rlc3Q=) on master @ [fce4d4723519bc4ca6e9ef5da0ae19960c84752c](https://github.com/cockroachdb/cockroach/commits/fce4d4723519bc4ca6e9ef5da0ae19960c84752c):


```
=== RUN   TestTenantStreamingPauseOnPermanentJobError
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingPauseOnPermanentJobError341685452
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:255: condition failed to evaluate within 45s: from testutils.go:267: Unexpected status failed
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingPauseOnPermanentJobError341685452
--- FAIL: TestTenantStreamingPauseOnPermanentJobError (53.73s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=29</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingPauseOnPermanentJobError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35812

--- a/pkg/ccl/streamingccl/replicationtestutils/testutils.go
+++ b/pkg/ccl/streamingccl/replicationtestutils/testutils.go
@@ -263,6 +263,7 @@ ORDER BY created DESC LIMIT 1`, c.Args.DestTenantName)
 			payload := jobutils.GetJobPayload(c.T, c.DestSysSQL, retentionJobID)
 			require.Contains(c.T, payload.Error, "replication stream")
 			require.Contains(c.T, payload.Error, "timed out")
+			return nil
 		}
 		return errors.Newf("Unexpected status %s", status)
 	})

